javascripts/discourse/components/discourse-tag-banners.js

```diff
@@ -0,0 +1,86 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+
+export default class DiscourseTagBanners extends Component {
+  @service store;
+  @service router;
+  @service site;
+  @tracked loaded = false;
+  @tracked tag = null;
+  @tracked formattedTagName = "";
+  @tracked formattedAdditionalTagNames = "";
+  @tracked additionalClass = "";
+  @tracked tagDescription = "";
+  @tracked isVisible = false;
+
+  constructor() {
+    super(...arguments);
+    this.router.on("routeDidChange", this, this.routeChanged);
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.router.off("routeDidChange", this, this.routeChanged);
+  }
```

Comment:
Would probably be cleaner to have:

```
<div class="tag-title-header" {{did-insert this.setupRouteChanged}} {{will-destroy this.teardownRouteChanged}}>
```

Reply:
Hmm Im not sure... Im wondering if we couldn't just do a computed property based on router currentRoute name, and then have a did-insert in the div inside for the `getTagInfo`... 🤔 

Let me know if it's not clear, I can try to look at this

Reply:
Making `isVisible` a getter which checks `this.router.currentRoute` would be great 👌 

The challenge is going to be triggering the `getTagInfo` ajax call at the right time. We can't use observers on native getters (and even if we could, we wouldn't want to). Another option would be firing the ajax request from a getter, but that feels kinda hacky.

Honestly I think an event listener might be the cleanest way to trigger an ajax request 🤔

Reply:
That essentially means "if there's a DOM change to this element, run `this.shouldRender`". But `this.shouldRender` is the thing which triggers the DOM update, so I think we have a bit of a chicken/egg problem here. Also as Kris mentioned, these hooks only work if the DOM element is actually being rendered, and in this case the entire DOM is wrapped inside an `{{#if shouldDisplay}}`.

Reply:
> Also I don't think this is correct:
> What it means is run this.getTagInfo if this.shouldRender changes

Oh you're absolutely right, my bad 👌